crates/matrix-sdk-common/src/linked_chunk/mod.rs

```diff
@@ -1074,6 +1074,15 @@ pub enum ChunkContent<Item, Gap> {
     Items(Vec<Item>),
 }
 
+impl<Item: Clone, Gap: Clone> Clone for ChunkContent<Item, Gap> {
```

Comment:
Doesn't it work with `#[derive(Clone)]` automatically? I thought `derive(Clone)` was adding the bounds for generic parameters automatically.

Reply:
Tested here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=477cdd4969b9e85bb263009447a6acd3

See the macro expand tool result:

```rust
#[derive(Clone)]
struct Foo<A, B> {
    a: A,
    b: B,
}

struct NotClone;

fn test<T: Clone>(value: T) {}

fn main() {
    test(Foo { a: 1, b: 2});
    test(Foo { a: 1, b: NotClone });
}
```

becomes:

```rust
struct Foo<A, B> {
    a: A,
    b: B,
}
#[automatically_derived]
impl<A: ::core::clone::Clone, B: ::core::clone::Clone> ::core::clone::Clone
    for Foo<A, B> {
    #[inline]
    fn clone(&self) -> Foo<A, B> {
        Foo {
            a: ::core::clone::Clone::clone(&self.a),
            b: ::core::clone::Clone::clone(&self.b),
        }
    }
}

// …
```

Reply:
nice